source/schemas/types.txt

```diff
@@ -509,3 +509,124 @@ with the ``title`` field set to the embedded object's type name:
+schema must a **required** ``type`` field, of type string, which is always set 
+to "Point".
+
+You must also provide a ``coordinates`` field, which is an array of doubles. 
+The ``coordinates`` array must contain  at least 2 double values, and may 
+contain a third:
+    
+- The first double is the longitude of the point, between -180 and 180.
+- The second is the latitude, between -90 and 90.
+- The optional third value represents the elevation/altitude of the point, 
+  in meters. Atlas ignores this value when performing queries. 
+
+.. note::
+
+   You cannot set an array as required in a Device Sync schema, so the Sync 
+   server checks that the coordinates field is present and meets the 
+   requirements.
+
+To store a GeoPoint in Atlas for later querying with :ref:`Atlas Device Sync <sync>`, 
+the following rules also apply to the GeoPoint object:
+
+- It must be of type ``object``
+- It must be embedded within another type
+- Additional properties are allowed within the ``coordinates`` array, as long as 
+  they are doubles.
+- Additional properties in the GeoPoint object are also allowed.
+ 
+The following is the Device Sync schema of an object with an embedded GeoPoint 
+property named "location":
+
+.. code-block:: json
+   :copyable: false
+
+   {
+     "title": "MyObject",
+     "properties": {
+        "location": {
+          "bsonType": "object",
+          "required": [ "type" ],
+          "properties": {
+            "type": {
+              "bsonType": "string",
+            },
+            "coordinates": {
+              "bsonType": "array",
+              "items": {
+                "bsonType": "double"
+              }
+            }
+          }
+        }
+      }
+   }
+
+The following code block shows a document that follows the schema. It has a 
+single embedded GeoPoint object named "location". Both the containing object 
+and the GeoPoint object contain an extra property.
+
+.. code-block:: json
+   :copyable: false
+   :emphasize-lines: 3-6
+
+   {
+     "_id": { "$oid": "65039d09fe4e46dddee31a3f" },
+     "location": {
+       "type": "Point",
+       "coordinates": [-122.4, 48.12, 23.0]
+      },
+     "another_optional_property": "geopoint name"
+   }
+
+GeoJSON Points and Device Sync Queries
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+GeoPoints are the only form of geospatial data that Device Sync can 
+query. The Device Sync server validates every embedded GeoPoint property with 
+the rules outlined above.
+
+If a device writes an object that fails validation, it will be reverted with a 
+:ref:`compensating write error <flexible-sync-errors>`. If a document that doesn't 
+meet these requirements is written directly to the MongoDB collection, the server 
+marks that document as unsyncable.
+
+Querying
+````````
+You can query a GeoPoint using :ref:`Realm Query Language (RQL) <rql>`, 
+using ``$geoWithin`` to do so. Although GeoPoints are the only stored type you 
```

Comment:
[nit] `$geoWithin` is a MongoDB operator. The RQL equivalent does not contain a dollar sign